crates/primitives/starknet/tests/state.rs

```diff
@@ -0,0 +1,60 @@
+use mp_starknet::state::*;
+use blockifier::execution::contract_class::ContractClass;
+use blockifier::state::state_api::{StateReader};
+use starknet_api::api_core::{ClassHash, ContractAddress, Nonce};
+use starknet_api::hash::StarkFelt;
+use starknet_api::state::StorageKey;
+
+#[test]
+fn test_get_storage_at() {
+    let mut state = DictStateReader::default();
+
+    let address = ContractAddress::default();
+    let key = StorageKey::default();
+    let value = StarkFelt::default();
+    let storage_key = (address, key);
+
+    state.storage_view.insert(storage_key, value);
+
+    let result = state.get_storage_at(address, key).unwrap();
+    assert_eq!(result, value);
+}
+
+#[test]
+fn test_get_nonce_at() {
+    let mut state = DictStateReader::default();
+
+    let address = ContractAddress::default();
+    let nonce = Nonce::default();
+
+    state.address_to_nonce.insert(address, nonce);
+
+    let result = state.get_nonce_at(address).unwrap();
+    assert_eq!(result, nonce);
+}
+
+#[test]
+fn test_get_contract_class() {
```

Comment:
check the error case also: `None => Err(StateError::UndeclaredClassHash(*class_hash)),`